tutorials/nginx-reverse-proxy-docker.md

```diff
@@ -0,0 +1,392 @@
+1.  Give the IP address a name, such as "reverse-proxy".
+1.  Click the **Create** button to create the Compute Engine instance.
+
+## Set up some domains for your instance
+
+Create multiple [A type DNS
+records](https://en.wikipedia.org/wiki/List_of_DNS_record_types) for various
+domains/subdomains on your DNS provider pointing at the external IP address for
+your new instance.
+
+For example, in [Google Domains](https://domains.google.com/registrar), open
+**DNS** for your domain, scroll to [Custom resource
+records](https://support.google.com/domains/answer/3251147) and add an **A**
+type record. The name "@" corresponds to the root of your domain or you can
+change it to a subdomain, such as "a" and "b".
+
+This tutorial will assume you have two subdomains with A records:
+
+- a.example.com
+- b.example.com
+
+## Deploying your web applications
+
+The reverse proxy will work for any web app running in Docker. This tutorial
+hosts two different static websites as an example.
+
+1.  SSH into your Compute Engine instance by clicking the **SSH** button on the
+    [instances page](https://console.cloud.google.com/compute/instances).
+1.  Create directory for "site A."
+
+        mkdir site-a
+        cd site-a
+
+1.  Configure a Docker image for "site A." This tutorial uses
+    [docker-nginx](https://github.com/KyleAMathews/docker-nginx) to host a
+    static website.
+
+        echo "FROM kyma/docker-nginx
+        COPY src/ /var/www
+        CMD 'nginx'" > Dockerfile
+
+1.  Make the static website pages.
+
+        mkdir src
+        echo "Hello from site A" > src/index.html
+
+1.  Build the Docker image.
+
+        docker build -t site-a .
+
+1.  Run the Docker container.
+
+        docker run -d --name site-a -p 80:80 site-a
+
+1.  View the running website at http://EXTERNAL_IP_ADDRESS,
+    http://a.example.com, or http://b.example.com.
+
+1.  Stop the container.
+
+        docker stop site-a
+
+1.  Remove the container.
+
+        docker rm site-a
+
+You now have a static website which you can run in a Docker container. Repeat
+these steps to get a second site set up.
+
+1.  Create directory for "site B."
+
+        cd
+        mkdir site-b
+        cd site-b
+
+1.  Configure a Docker image for "site B."
+
+        echo "FROM kyma/docker-nginx
+        COPY src/ /var/www
+        CMD 'nginx'" > Dockerfile
+
+1.  Make the static website pages.
+
+        mkdir src
+        echo "Hello from site B" > src/index.html
+
+1.  Build the Docker image. This time will be faster because the base image is
+    cached.
+
+        docker build -t site-b .
+
+1.  Run the Docker container.
+
+        docker run -d --name site-b -p 80:80 site-b
+
+1.  View the running website at http://EXTERNAL_IP_ADDRESS,
+    http://a.example.com, or http://b.example.com.
+
+1.  Stop the container.
+
+        docker stop site-b
+
+1.  Remove the container.
+
+        docker rm site-b
+
+Now you have two web apps, but since they both listen to port 80, you can't run
```

Comment:
Should I just cut to the final step?

I originally had this separated into three phases: web host with Docker, proxy, SSL so that each step could be tested separately.

I know I wouldn't be able to debug if something goes wrong if I just skipped to the full SSL reverse proxy.